modules/WindowsScriptExtension_v1/main.tf

```diff
@@ -10,25 +10,28 @@ NOTES
 - TF Function: templatefile Reads a file, and replaces text with other text.
 */
 
-locals {
-  scriptName     = var.scriptName                                      #This cannot be the full path.  Must be script name only.
-  scriptRendered = filebase64("${var.scriptPath}/${local.scriptName}") #Full path.
-  # use templatefile() to parse script parameters
-  #ifTemplateFile = base64encode(templatefile("${path.module}/${local.scriptName}", {}))
-  commandToExecute = jsonencode({
-    commandToExecute = "powershell -command \"[System.Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String('${local.scriptRendered}')) | Out-File -filepath ${local.scriptName}\" && powershell -ExecutionPolicy Unrestricted -File ${local.scriptName}"
-  })
-}
-
 # settings block will look like
 
-resource "azurerm_virtual_machine_extension" "ScriptExtension" {
-  name                 = "ScriptExtension"
-  virtual_machine_id   = var.virtual_machine_id
-  publisher            = "Microsoft.Compute"
-  type                 = "CustomScriptExtension"
-  type_handler_version = "1.10"
+data "azurerm_storage_account" "test" {
+  name                = var.sa_name
+  resource_group_name = var.storage_account_rg
+}
 
-  protected_settings = local.commandToExecute
+resource "azurerm_virtual_machine_extension" "ScriptExtension" {
+  name                       = "ScriptExtension"
+  virtual_machine_id         = var.virtual_machine_id
+  publisher                  = "Microsoft.Compute"
+  type                       = "CustomScriptExtension"
+  type_handler_version       = "1.10"
+  auto_upgrade_minor_version = true
+  protected_settings         = <<PROTECTED_SETTINGS
+    {         
+            "fileUris": ["https://ansibleteststrgaccnt.blob.core.windows.net/winrmext/InitializeDisks.ps1",

```

Comment:
Can you make the path to the files a variable?  Then we can move them around.

Reply:
Updated file path as a variable.